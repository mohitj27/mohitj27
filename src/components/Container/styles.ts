/**
 * Container component styles.
 */
import { Styles } from 'react-jss';
import {
  breakpoints,
  midBlue,
  textLightBlue,
  textMid,
  mSize,
} from '../../theme';

export default {
  root: {
    maxWidth: '960px',
    padding: '0 20px',
    marginBottom: '200px',
    '& a:hover': {
      color: textMid,
    },
    '& li': {
      listStyle: 'square',
    },
    [`@media screen and (max-width: ${breakpoints.desktop})`]: {
      marginLeft: 0,
    },
  },
  content: {
    color: textLightBlue,
    fontSize: mSize,
    '& h3': {
      color: midBlue,
    },
    '& ul': {
      '& li': {
        paddingBottom: '4px',
      },
      [`@media screen and (max-width: ${breakpoints.desktop})`]: {
        paddingLeft: '20px',
      },
    },
  },
} as Styles;
