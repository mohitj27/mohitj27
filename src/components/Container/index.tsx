/**
 * Container components used in tabs.
 */
import React from 'react';
import { createUseStyles } from 'react-jss';
import SEO, { SEOProps } from '../SEO';
import styles from './styles';

const useStyles = createUseStyles(styles);

export interface ContainerProps {
  seo?: SEOProps;
}

const Container: React.FC<ContainerProps> = ({ seo, children }) => {
  const classes = useStyles();

  return (
    <>
      {!!seo && <SEO {...seo} />}
      <div className={classes.root}>
        <div className={classes.content}>{children}</div>
      </div>
    </>
  );
};

export default Container;
