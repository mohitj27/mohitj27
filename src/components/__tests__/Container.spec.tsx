import React from 'react';
import { render } from '@testing-library/react';
import Container from '../Container';

describe('<Container />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <Container
        title="Test Title"
        subtitle="Test Subtitle"
        seo={{
          title: 'Test SEO Title',
          description: 'Test SEO Description',
        }}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
