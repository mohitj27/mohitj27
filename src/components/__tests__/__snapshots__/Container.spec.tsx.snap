// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root-0-2-1"
  >
    <div
      class="content-0-2-2"
    />
  </div>
</DocumentFragment>
`;
